File: src/ledgerloop/report.py
"""Render a reconciliation as a Markdown report."""

from __future__ import annotations

from .models import Reconciliation


def render_report(rec: Reconciliation) -> str:
    s = rec.summary
    return (
        "# Reconciliation Report\n\n"
        f"- Matched: {s['matched']}\n"
        f"- Unmatched (left): {s['unmatched_left']}\n"
        f"- Unmatched (right): {s['unmatched_right']}\n"
    )

File: src/webapp/report.js
// Summary + Markdown report for a reconciliation result.
window.LedgerLoop = window.LedgerLoop || {};

window.LedgerLoop.summarize = function summarize(rec) {
  const matched = rec.matched.length;
  const ul = rec.unmatchedLeft.length;
  const ur = rec.unmatchedRight.length;
  const total = matched + ul + ur;
  const rate = total === 0 ? 0 : Math.round((matched / total) * 100);
  return { matched, unmatchedLeft: ul, unmatchedRight: ur, rate };
};

window.LedgerLoop.toMarkdown = function toMarkdown(rec) {
  const s = window.LedgerLoop.summarize(rec);
  return [
    "# LedgerLoop Reconciliation Report",
    "",
    "- Matched: " + s.matched,
    "- Unmatched (A): " + s.unmatchedLeft,
    "- Unmatched (B): " + s.unmatchedRight,
    "- Match rate: " + s.rate + "%",
  ].join("\n");
};
